### ISSUE TITLE
\d source type

### ISSUE THREAD

**Issue by laurenarnett** [MaterializeInc/materialize#6578]
Just worked through the getting started tutorial. The `wikirecent` demo is great and illustrates Materialize's potential very well!

When I went to clean up from the demo by dropping the views created and my wikirecent source, I used `\d` to examine what all had been created, and was surprised to see the following output:
```
                        List of relations
   Schema   |                Name                 | Type  | Owner 
------------+-------------------------------------+-------+-------
.
.
.
public      | wikirecent                          | table | 
```
Unsure whether the type of `wikirecent` was a `source`, as I had used to create it, or if under the hood `wikirecent` is represented a `table`, as feels more familiar coming from Postgres, I first tried `DROP TABLE wikirecent;` which returned to me: `ERROR:  materialize.public.wikirecent is not of type TABLE`. Going to the Materialize docs, I found the command I was looking for was in fact `DROP SOURCE`, which was successful. 

I came across the conversation at [#4424](https://github.com/MaterializeInc/materialize/pull/4424), and just wanted to add to it from a beginner user point of view! I understand the idea behind treating sources/tables as equivalent here, but it may be helpful to have a separate source type listed to be symmetric with the separate `CREATE`, `SHOW`, `DROP`, etc. commands for source and table.

**Comment by rjnn** [comment]
Thanks @laurenarnett! This report is very appreciated, thank you for the suggestion. I can see how this would be confusing as well. I'm not entirely sure if there is a clean fix given I don't fully understand the intricacies of what `\d` translates into under the hood (cc @JLDLaughlin) - but at the very least we should improve the error message when you attempt `DROP TABLE wikirecent` (perhaps it can just spell it out entirely, i.e.: "Error: wikirecent is not of type TABLE, try DROP SOURCE wikirecent").

**Comment by benesch** [comment]
> Thanks @laurenarnett! This report is very appreciated, thank you for the suggestion. I can see how this would be confusing as well. I'm not entirely sure if there is a clean fix given I don't fully understand the intricacies of what `\d` translates into under the hood (cc @JLDLaughlin) - but at the very least we should improve the error message when you attempt `DROP TABLE wikirecent` (perhaps it can just spell it out entirely, i.e.: "Error: wikirecent is not of type TABLE, try DROP SOURCE wikirecent").

Yeah unfortunately our hands are a bit tied here. PostgreSQL doesn't have the concept of sources so there's not a whole lot we can do, aside from not presenting sources at all in `\d` (which really could be preferable! not sure!).

Improving the error message would be great, but is blocked on some of the work described in #5340.
